hpc/jvergeyn-vacc-strategies.py

```diff
@@ -0,0 +1,681 @@
+initN, Nc_home, Nc_work, Nc_schools, Nc_transport, Nc_leisure, Nc_others, Nc_total = model_parameters.get_interaction_matrices(dataset='willem_2012',intensity='all')
+# Define the number of age categories
+levels = initN.size
+
+initN, Nc_home_15, Nc_work_15, Nc_schools_15, Nc_transport_15, Nc_leisure_15, Nc_others_15, Nc_total_15 = model_parameters.get_interaction_matrices(dataset='willem_2012',intensity='more_15_min')
+initN, Nc_home_1hr, Nc_work_1hr, Nc_schools_1hr, Nc_transport_1hr, Nc_leisure_1hr, Nc_others_1hr, Nc_total_1hr = model_parameters.get_interaction_matrices(dataset='willem_2012',intensity='more_one_hour')
+
+Nc_all = {'total': Nc_total, 'home':Nc_home, 'work': Nc_work, 'schools': Nc_schools, 'transport': Nc_transport, 'leisure': Nc_leisure, 'others': Nc_others}
+Nc_15min = {'total': Nc_total_15, 'home': Nc_home_15, 'work': Nc_work_15, 'schools': Nc_schools_15, 'transport': Nc_transport_15, 'leisure': Nc_leisure_15, 'others': Nc_others_15}
+Nc_1hr = {'total': Nc_total_1hr, 'home': Nc_home_1hr, 'work': Nc_work_1hr, 'schools': Nc_schools_1hr, 'transport': Nc_transport_1hr, 'leisure': Nc_leisure_1hr, 'others': Nc_others_1hr}
+
+
+# ## Load publically available data from Sciensano
+
+df_sciensano = sciensano.get_sciensano_COVID19_data(update=False)
+df_sciensano['D_cum'] = df_sciensano.D_tot.cumsum()
+df_google = mobility.get_google_mobility_data(update=False, plot=False)
+
+
+# ## Load posterior parameter distributions of calibrated parameters
+
+# Load samples dictionaries
+with open('../data/interim/model_parameters/COVID19_SEIRD/calibrations/national/BE_4_prev_full_2021-01-30_WAVE2_GOOGLE.json', 'r') as fp:
+    samples_dict = json.load(fp)
+
+with open('../data/interim/model_parameters/COVID19_SEIRD/calibrations/national/initial_states_2020-09-01.json', 'r') as fp:
+    initial_states_sept = json.load(fp) 
+    
+# Set initial states of VE (vaccine elegible)
+initial_states_sept['VE'] = (np.array(initial_states_sept['S'])+
+                             np.array(initial_states_sept['R'])+
+                             np.array(initial_states_sept['E'])+
+                             np.array(initial_states_sept['I'])+
+                             np.array(initial_states_sept['A'])).squeeze()
+
+## Scenarios
+
+start_sim = '2020-09-01'
+end_sim = '2021-09-01'
+start_fig = '2021-01-01'
+warmup = 0
+n_samples = 100
+n_draws_per_sample = 1000
+
+Re_1feb = 0.958*1.4
+# current estimation of UK prevalence between 20 and 40% => take 30%
+current_UK = 0.3
+# 5-6 days incubation
+incubation_period = 5
+n_periods = 30/incubation_period
+# How much on January 1? (30 days ago)
+portion_new_strain_introduced = current_UK/(Re_1feb**n_periods) #0.001
+injection_day = (pd.Timestamp('2021-01-01') - pd.Timestamp(start_sim))/pd.Timedelta('1D')
+
+UL = 0.975
+LL = 0.025
+
+# doses per day (Regeringscommissariaat Corona, published by De Tijd)
+d = {}
+d['jan'] = 31765
+d['feb'] = 45897
+d['mar-apr'] = 128499
+d['may-aug'] = 78358
```

Comment:
Voor later: we moeten hier rechtstreeks de sciensano data kunnen inknallen tot present day, en het vaccinatieplan gebruiken in de toekomst. Er moet dus,

1. Een functie worden gedefinieerd om de vaccinaties te downloaden en in te laden.
2. Een vaccintiefunctie, analoog aan de policy functie met google (en `@lru_cache`) gemaakt worden.

Eventueel kan dan zelf onderscheid gemaakt worden tussen de vaccins en hun leakiness/efficientie. Ik weet niet of sciensano het vaccin dat gebruikt werd ter beschikbaar stelt in zijn dataset. Je kan de 'gemiddelde efficientie' van alle gezette spuiten berekenen in je vaccinatiefunctie, en die dan doorgeven aan je model. Bvb: 1 miljoen (2x) spuiten van Pfizer aan 90% efficientie en 1 miljoen spuiten van AstraZeneca aan 60% efficientie (getallen zijn illustratief), geeft een gemiddelde dekking van 75% procent. Die geef je dan op elke tijdstap door aan je model. Uiteraard klopt dit niet 100%, want als je eerst die miljoen spuiten zet van Pfizer en daarna die van AstraZeneca dan gaat dat gemiddelde misschien achterlopen ofzo. Evenuteel kan je de vaccinatieefficientie dan eeftijdsafhankelijk maken.

We zouden best na je conge eens overleggen omtrent die vaccinaties denk ik, er zijn veel leuke dingen die hier nog aan kunnen worden toegevoegd.